flaml/automl/task/generic_task.py

```diff
@@ -664,19 +590,19 @@ def decide_split_type(
             ), "GroupKFold requires groups to be provided."
             return split_type
 
-        elif self.is_ts_forecast():
-            assert split_type in ["auto", "time"]
-            assert isinstance(
-                fit_kwargs.get("period"),
-                int,  # NOTE: _decide_split_type is before kwargs is updated to fit_kwargs_by_estimator
-            ), f"missing a required integer 'period' for '{TS_FORECAST}' task."
-            if fit_kwargs.get("group_ids"):
-                # TODO (MARK) This will likely not play well with the task class
-                self.name = TS_FORECASTPANEL
-                assert isinstance(
-                    fit_kwargs.get("group_ids"), list
-                ), f"missing a required List[str] 'group_ids' for '{TS_FORECASTPANEL}' task."
-            return "time"
+        # elif self.is_ts_forecast():
```

Comment:
It seems this part is moved to the `decide_split_type` func of `TimeSeriesTask`. Are we assuming an instance of `GenericTask` is never a ts forecast task (thus, `self.is_ts_forecast` is always False)?  Even if so, I think it is better to add a warning or assertion msg here instead of directly removing it.

Reply:
> Are we assuming an instance of GenericTask is never a ts forecast task?

Yes, we are. Will add an assertion.